tests/config_test.cc

```diff
@@ -0,0 +1,160 @@
+#include "cris/core/config/config.h"
+
+#include <gtest/gtest.h>
+
+#include <fmt/core.h>
+
+#include <chrono>
+#include <cstdint>
+#include <cstdio>
+#include <filesystem>
+#include <fstream>
+#include <thread>
+
+namespace cris::core {
+
+constexpr double kEps = 1e-6;
+
+namespace fs = std::filesystem;
+
+class ConfigTest : public testing::Test {
+   public:
+    ConfigTest()
+        : testing::Test()
+        , test_config_dir_(std::tmpnam(nullptr))
+        , test_config_path_(test_config_dir_ / "test.config") {
+        EXPECT_TRUE(fs::create_directory(test_config_dir_));
+    }
+
+    ~ConfigTest() { fs::remove_all(test_config_dir_); }
```

Comment:
This is dangerous. Always do some sanity check right before removal to ensure you're touching the right thing:
- is dir.
- Name is not empty.
- Name looks good.